Products/CMFPlone/image_scales/adapters.py

```diff
@@ -0,0 +1,142 @@
+from Acquisition import aq_inner
+from plone.dexterity.interfaces import IDexterityContent
+from plone.dexterity.utils import iterSchemata
+from plone.namedfile.interfaces import INamedImageField
+from plone.registry.interfaces import IRegistry
+from Products.CMFPlone.image_scales.interfaces import IImageScalesAdapter
+from Products.CMFPlone.image_scales.interfaces import IImageScalesFieldAdapter
+from Products.CMFPlone.interfaces import IImagingSchema
+from zope.component import adapter
+from zope.component import getMultiAdapter
+from zope.component import getUtility
+from zope.component import queryMultiAdapter
+from zope.interface import implementer
+from zope.interface import Interface
+from zope.schema import getFields
+
+
+@implementer(IImageScalesAdapter)
+@adapter(IDexterityContent, Interface)
+class ImageScales:
+    def __init__(self, context, request):
+        self.context = context
+        self.request = request
+
+    def __call__(self):
+        obj = aq_inner(self.context)
+        res = {}
+        for schema in iterSchemata(self.context):
+            for name, field in getFields(schema).items():
+                # serialize the field
+                serializer = queryMultiAdapter(
+                    (field, obj, self.request), IImageScalesFieldAdapter
+                )
+                if serializer:
+                    scales = serializer()
+                    if scales:
+                        res[name] = scales
+        return res
+
+
+def _split_scale_info(allowed_size):
+    name, dims = allowed_size.split(" ")
+    width, height = list(map(int, dims.split(":")))
+    return name, width, height
+
+
+def _get_scale_infos():
+    """Returns list of (name, width, height) of the available image scales."""
+    registry = getUtility(IRegistry)
+    imaging_settings = registry.forInterface(IImagingSchema, prefix="plone")
+    allowed_sizes = imaging_settings.allowed_sizes
+    return [_split_scale_info(size) for size in allowed_sizes]
+
+
+@implementer(IImageScalesFieldAdapter)
+@adapter(INamedImageField, IDexterityContent, Interface)
+class ImageFieldScales:
```

Comment:
The specific field adapter should go to plone.namedfile. It was never planned to keep it here, in Bonn we agreed to put it here for rapid prtotyping. cc/ @cekk